addons/l10n_it_edi/models/res_config_settings.py

```diff
@@ -1,98 +1,45 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import api, models, fields, _
-from odoo.exceptions import UserError
+from odoo import _, api, models, fields
+from odoo.exceptions import ValidationError
 
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
-    is_edi_proxy_active = fields.Boolean(compute='_compute_is_edi_proxy_active')
-    l10n_it_edi_proxy_current_state = fields.Char(compute='_compute_l10n_it_edi_proxy_current_state')
-    l10n_it_edi_sdicoop_register = fields.Boolean(compute='_compute_l10n_it_edi_sdicoop_register', inverse='_set_l10n_it_edi_sdicoop_register_demo_mode')
-    l10n_it_edi_sdicoop_demo_mode = fields.Selection(
-        [('demo', 'Demo'),
-         ('test', 'Test (experimental)'),
-         ('prod', 'Official')],
-        compute='_compute_l10n_it_edi_sdicoop_demo_mode',
-        inverse='_set_l10n_it_edi_sdicoop_register_demo_mode',
+    l10n_it_edi_register = fields.Boolean(
+        compute='_compute_l10n_it_edi_register',
+        inverse='_set_l10n_it_edi_mode',
         readonly=False)
+    l10n_it_edi_mode = fields.Selection(
+        selection=[('demo', 'Demo'), ('test', 'Test (experimental)'), ('prod', 'Official')],
+        compute="_compute_l10n_it_edi_mode", inverse="_set_l10n_it_edi_mode", readonly=False)
+    l10n_it_edi_mode_old = fields.Selection(related="company_id.l10n_it_edi_mode")
```

Comment:
We should try to get rid of this and simplify.  Also now in master, we can store fields if necessary and probably, it is still simpler to have the selection on the company.